app/frontend/src/search/client.js

```diff
@@ -0,0 +1,26 @@
+import algoliasearch from 'algoliasearch';
+import instantsearch from 'instantsearch.js';
+
+import { convertMilesToMetres } from './utils';
+
+// This is the public API key which can be safely used in your frontend code.
+// This key is usable for search queries and list the indices you've got access to.
+export const search = algoliasearch('QM2YE0HRBW', '20b88d28047d5e3d60437993ad3d9c50');
+
+export const searchClient = indexName => instantsearch({
+    indexName: indexName,
+    searchClient: search,
+    searchFunction(helper) {
+        if (document.querySelector('#location').dataset.coordinates) {
+            helper.state.aroundLatLng = document.querySelector('#location').dataset.coordinates;
+        }
+
+        if (document.querySelector('#location').dataset.radius) {
+            helper.state.aroundRadius = convertMilesToMetres(document.querySelector('#location').dataset.radius);
+        }
+
```

Comment:
these 8 lines are a bit ugly and i plan to abstract this and put tests behind it

Reply:
I'm guessing you're planning to abstract that `dataset` attribute on `querySelector` out into `const`?